### PR TITLE
feat: microapps media API support

### PR DESCRIPTION
**Pull request by ekwonye-richard**
## Context

There is a new integration with google where we need to render transferwise.com inside https://microapps.google.com domain. There will be an android native application with a webview which will open transferwise.com inside.

We need to use Media API exposed by microapps for file upload, hence we need to make this the preferred option when the upload component is rendered inside the supported domain.

## Changes

Use Media API for file upload when the upload component is rendered inside the supported domain.

## Checklist

- [ ] All changes are covered by tests

### CODE REVIEW

File: src/forms/upload/upload.controller.js [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+    MediaApiService
   ) {
+    this.$window = $window;
     this.$element = $element;
     this.$attrs = $attrs;
     this.FileValidationService = FileValidationService;
+    this.MediaApiService = MediaApiService;
 
     this.isProcessing = false;
+    this.checkMediaUploadSupport = this.checkMediaUploadSupport.bind(this);
```
Comment (StevePole) [inline src/forms/upload/upload.controller.js:17]: I feel like we could encapsulate all of this logic within the capture card.  It's really only a concern for the component responsible for capturing the file, and it doesn't appear to have any impact on this controller/template, except for passing down the state to the capture card.

File: src/forms/upload/services/media-api.service.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import angular from 'angular';
+
+class MediaApiService {
```
Comment (StevePole) [inline src/forms/upload/services/media-api.service.js:3]: I like how you encapsulated this stuff in the service 👍

File: src/forms/upload/capture-card/capture-card.controller.js [hunk 3]
```diff
@@ -38,8 +51,21 @@ class Controller {
     const file = files[0];
     this.onFileCapture({ file });
   }
+
+  captureFromMedia() {
```
Comment (StevePole) [inline src/forms/upload/capture-card/capture-card.controller.js:55]: Perhaps this logic could be encapsulated in the service as well?

File: src/forms/upload/upload.controller.js (hunk 1 above)
Comment (ekwonye-richard) [reply src/forms/upload/upload.controller.js:17]: Yes, I agree with you on this. I'll move this to card capture

File: src/forms/upload/capture-card/capture-card.controller.js (hunk 3 above)
Comment (ekwonye-richard) [reply src/forms/upload/capture-card/capture-card.controller.js:55]: Yes, we can. I'll move it too.

File: src/forms/upload/capture-card/capture-card.controller.js [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
+    if (this.$window.document.readyState === 'complete') {
+      this.checkMediaUploadSupport();
+    } else {
+      this.$window.addEventListener('load', this.checkMediaUploadSupport);
+    }
   }
 
   $onChanges(changes) {
-    if (changes.icon || changes.isLiveCameraUpload) {
+    if ((changes.icon || changes.isLiveCameraUpload) && !this.isMediaUpload) {
```
Comment (StevePole) [inline src/forms/upload/capture-card/capture-card.controller.js:18]: I don't see a problem, but interested why this needs to change?

File: src/forms/upload/services/media-api.service.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
```
Comment (StevePole) [inline src/forms/upload/services/media-api.service.js:3]: Could we perhaps add a doc block to explain what this service is all about.  I worry that someone coming to this fresh would have no context for this.

File: src/forms/upload/capture-card/capture-card.controller.js (hunk 4 above)
Comment (ekwonye-richard) [reply src/forms/upload/capture-card/capture-card.controller.js:18]: If media upload is supported it's good we keep the default icon as camera.

Comment (ekwonye-richard) [reply src/forms/upload/capture-card/capture-card.controller.js:18]: On a second thought, I should make this check only when icon isn't changed.

File: src/forms/upload/services/media-api.service.js (hunk 5 above)
Comment (ekwonye-richard) [reply src/forms/upload/services/media-api.service.js:3]: I will add this

**Review by StevePole — APPROVED** [review]
Nice tests 👍 

There may be a little repetition with the upload tests.  I think previously we were treating the whole upload as the testable 'unit', hence the card itself didn't have tests.  Though I'm sure what you've added is more thorough anyway.

**Pull request merged**